diff --git a/src/Normalizer/RelationshipItem.php b/src/Normalizer/RelationshipItem.php
index 41e4182..79c5c21 100644
--- a/src/Normalizer/RelationshipItem.php
+++ b/src/Normalizer/RelationshipItem.php
@@ -70,7 +70,8 @@ class RelationshipItem {
         $host_entity->bundle()
       )->getRelatableResourceTypes()[$parent->getPropertyName()];
       if (count($relatable_resource_types) !== 1) {
-        throw new \RuntimeException('Relationships to virtual resources are possible only if a single resource type is relatable.');
+        \Drupal::logger('worksafe_jsonapi')->error('The node being returned on this jsonapi endpoint should be resaved as it has bad data.');
+        // throw new \RuntimeException('Relationships to virtual resources are possible only if a single resource type is relatable.');
       }
       $this->targetResourceType = reset($relatable_resource_types);
     }
